### PR TITLE
Now uses stamp

### PR DESCRIPTION
**Pull request by tnickelsen**

### CODE REVIEW

File: domains/certificates/Query.API/API.IntegrationTests/Factories/QueryApiWebApplicationFactory.cs [hunk 1]
```diff
@@ -104,7 +111,14 @@ protected override void ConfigureWebHost(IWebHostBuilder builder)
             });
 
             // Remove MeasurementsSyncerWorker
-            services.Remove(services.First(s => s.ImplementationType == typeof(MeasurementsSyncerWorker)));
+            if (!MeasurementsSyncEnabled)
+                services.Remove(services.First(s => s.ImplementationType == typeof(MeasurementsSyncerWorker)));
+
+            if (measurementsClient != null)
```
Comment (ckr123) [inline domains/certificates/Query.API/API.IntegrationTests/Factories/QueryApiWebApplicationFactory.cs:117]: Why is this needed?

File: domains/certificates/Query.API/API/MeasurementsSyncer/MeasurementsSyncService.cs [hunk 2]
```diff
@@ -3,39 +3,50 @@
...
         this.slidingWindowState = slidingWindowState;
         this.measurementsClient = measurementsClient;
-        this.bus = bus;
         this.slidingWindowService = slidingWindowService;
         this.measurementSyncMetrics = measurementSyncMetrics;
+        this.stampClient = stampClient;
     }
 
-    public async Task FetchAndPublishMeasurements(string meteringPointOwner, MeteringPointTimeSeriesSlidingWindow slidingWindow,
+    public async Task FetchAndPublishMeasurements(string meteringPointOwner,
```
Comment (ckr123) [inline domains/certificates/Query.API/API/MeasurementsSyncer/MeasurementsSyncService.cs:44]: Can you take syncinfo as a parameter? instead of 5 of these parameters, so syncinfo and slidingwindow?

File: domains/certificates/Query.API/API/MeasurementsSyncer/MeasurementsSyncerWorker.cs [hunk 3]
```diff
@@ -81,6 +82,22 @@ private async Task PerformPeriodicTask(CancellationToken stoppingToken)
         }
     }
 
+    private async Task Sleep(CancellationToken cancellationToken)
+    {
+        if (options.SleepType == MeasurementsSyncerSleepType.Hourly)
```
Comment (ckr123) [inline domains/certificates/Query.API/API/MeasurementsSyncer/MeasurementsSyncerWorker.cs:87]: Why these different sleeptypes?

File: domains/certificates/Query.API/API/MeasurementsSyncer/MeteringPointSyncInfo.cs [hunk 4]
```diff
@@ -1,5 +1,12 @@
 using System;
+using DataContext.ValueObjects;
 
 namespace API.MeasurementsSyncer;
 
-public record MeteringPointSyncInfo(string GSRN, DateTimeOffset StartSyncDate, string MeteringPointOwner);
+public record MeteringPointSyncInfo(string GSRN,
```
Comment (ckr123) [inline domains/certificates/Query.API/API/MeasurementsSyncer/MeteringPointSyncInfo.cs:6]: You could wrap all these arguments into the Contract model

File: domains/certificates/README.md [hunk 5]
```diff
@@ -65,5 +65,4 @@ This is the commands for generating the migration SQL script for the API project
 
 ```shell
 dotnet ef migrations script --idempotent --project Query.API/API --output migrations/API.sql
-cp migrations/API.sql migrations/Worker.sql
```
Comment (ckr123) [inline domains/certificates/README.md:68]: Dont we need the API.sql anymore?

File: domains/certificates/Shared/DataContext/Migrations/20240723091045_ModifiedContractsToUseRecipientIdAndRemovedCertificates.cs [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+namespace DataContext.Migrations
+{
+    /// <inheritdoc />
+    public partial class ModifiedContractsToUseRecipientIdAndRemovedCertificates : Migration
+    {
+        /// <inheritdoc />
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.Sql("DELETE FROM public.\"Contracts\";");
```
Comment (ckr123) [inline domains/certificates/Shared/DataContext/Migrations/20240723091045_ModifiedContractsToUseRecipientIdAndRemovedCertificates.cs:14]: Didnt we agree on a migrations period so we dont delete peoples contracts, we just add recipendId?

File: domains/certificates/Shared/DataContext/ApplicationDbContext.cs [hunk 7]
```diff
@@ -47,8 +42,6 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
     }
 
     public DbSet<CertificateIssuingContract> Contracts { get; set; }
-    public DbSet<ProductionCertificate> ProductionCertificates { get; set; }
```
Comment (ckr123) [inline domains/certificates/Shared/DataContext/ApplicationDbContext.cs:50]: We dont need these to hold state anymore?

File: domains/certificates/Query.API/API.IntegrationTests/Factories/QueryApiWebApplicationFactory.cs (hunk 1 above)
Comment (tnickelsen) [reply domains/certificates/Query.API/API.IntegrationTests/Factories/QueryApiWebApplicationFactory.cs:117]: I made it possible to mock the client. And it is only set here if you do that

File: domains/certificates/Query.API/API/MeasurementsSyncer/MeasurementsSyncerWorker.cs (hunk 3 above)
Comment (tnickelsen) [reply domains/certificates/Query.API/API/MeasurementsSyncer/MeasurementsSyncerWorker.cs:87]: Only breaks if we don't set it. We need it to run the MeasurementsSyncer faster for integration tests

File: domains/certificates/Query.API/API/MeasurementsSyncer/MeteringPointSyncInfo.cs (hunk 4 above)
Comment (tnickelsen) [reply domains/certificates/Query.API/API/MeasurementsSyncer/MeteringPointSyncInfo.cs:6]: Not sure what you mean

File: domains/certificates/README.md (hunk 5 above)
Comment (tnickelsen) [reply domains/certificates/README.md:68]: Not the worker.sql

File: domains/certificates/Shared/DataContext/Migrations/20240723091045_ModifiedContractsToUseRecipientIdAndRemovedCertificates.cs (hunk 6 above)
Comment (tnickelsen) [reply domains/certificates/Shared/DataContext/Migrations/20240723091045_ModifiedContractsToUseRecipientIdAndRemovedCertificates.cs:14]: What do you mean by 'migrations period'?
We agreed to clear the contracts so that I can add the recipientId column

File: domains/certificates/Shared/DataContext/ApplicationDbContext.cs (hunk 7 above)
Comment (tnickelsen) [reply domains/certificates/Shared/DataContext/ApplicationDbContext.cs:50]: No we decided to drop them

File: domains/certificates/Query.API/API/MeasurementsSyncer/MeasurementsSyncService.cs (hunk 2 above)
Comment (tnickelsen) [reply domains/certificates/Query.API/API/MeasurementsSyncer/MeasurementsSyncService.cs]: done

File: domains/certificates/Query.API/API/MeasurementsSyncer/MeasurementsSyncService.cs [hunk 8]
```diff
@@ -60,7 +56,12 @@ public class MeasurementsSyncService
 
             if (measurementsToPublish.Any())
             {
-                await IssueCertificate(measurementsToPublish, meteringPointType, gridArea, recipientId, technology, stoppingToken);
+                await IssueCertificates(measurementsToPublish,
```
Comment (ckr123) [inline domains/certificates/Query.API/API/MeasurementsSyncer/MeasurementsSyncService.cs:59]: These could also just be syncInfo object :)

File: domains/certificates/Query.API/API/MeasurementsSyncer/MeteringPointSyncInfo.cs (hunk 4 above)
Comment (ckr123) [reply domains/certificates/Query.API/API/MeasurementsSyncer/MeteringPointSyncInfo.cs:6]: It looks like all these parameters come from the "oldestContract" in contractState.cs 
So you could just make the method like this `MeteringPintSyncInfo(Contract contract)`

Comment (tnickelsen) [reply domains/certificates/Query.API/API/MeasurementsSyncer/MeteringPointSyncInfo.cs:6]: That is too much info coming into syncInfo. Contract holds more info than needed

File: domains/certificates/Query.API/API/MeasurementsSyncer/MeasurementsSyncService.cs (hunk 8 above)
Comment (tnickelsen) [reply domains/certificates/Query.API/API/MeasurementsSyncer/MeasurementsSyncService.cs:59]: done

File: domains/certificates/Query.API/API/MeasurementsSyncer/MeteringPointSyncInfo.cs (hunk 4 above)
Comment (ckr123) [reply domains/certificates/Query.API/API/MeasurementsSyncer/MeteringPointSyncInfo.cs:6]: Maybe i am missing some use cases, but IssuingContracts contains 
            Id = contract.Id,
            GSRN = contract.GSRN,
            StartDate = contract.StartDate.ToUnixTimeSeconds(),
            EndDate = contract.EndDate?.ToUnixTimeSeconds(),
            Created = contract.Created.ToUnixTimeSeconds(),
            MeteringPointType = contract.MeteringPointType,
            Technology = contract.Technology
            
           Why do we have this meteringpointsyncinfo model, it looks like a copy of issuingcontract?

Comment (tnickelsen) [reply domains/certificates/Query.API/API/MeasurementsSyncer/MeteringPointSyncInfo.cs:6]: It's there for two different purposes. If contracts gets more logic we would have to expand sync infos as well. Besides sync info dont need created and endDate

**Review by ckr123 — APPROVED** [review]
